### ISSUE TITLE
Issue with build

### ISSUE THREAD

**Issue by yashha** [Rostlab/JS16_ProjectF#203]
https://travis-ci.org/Rostlab/JS16_ProjectF/builds/119496162

**Comment by gyachdav** [comment]
Do you guys have an E representative for your project?

Shouldn't they take care of all your build and config issues?

@sacdallago please intervene and assign someone to take care of all test,build,deploy issues that F is having.

**Comment by yashha** [comment]
I thought that partly my job because I set it up, but we have one @TheZoker is doing testing for us. Rostlab/JS16_ProjectE#19

**Comment by gyachdav** [comment]
@yashha you need to learn to delegate :smile: 

Make sure to involve @TheZoker as much as possible

**Comment by TheZoker** [comment]
Yes I'm here to help!
The issue is closed, so is this fixed?

**Comment by sacdallago** [comment]
:D Apparently I still assigned you just in case

**Comment by TheZoker** [comment]
Yes that's good :)
I currently try to understand, what was the issue here: https://github.com/Rostlab/JS16_ProjectF/commit/1f33dc2ec3d4accd6bbe9e43d4abf54da5f46206

Seems like a valid package.json to me :confused:

**Comment by yashha** [comment]
I think the problem was the some dependecies were not loading
But i don't know :)
I changed it to this and now it works

**Comment by sacdallago** [comment]
actually not separating devDependencies and dependencies is pretty, pretty bad practice.

Try to figure out if you can sort this out without resorting on patches @TheZoker please

**Comment by TheZoker** [comment]
Ok I contacted the travis support, hopefully I get an answer soon.

**Comment by TheZoker** [comment]
I got a answer, but no solution yet. I will keep you up2date

**Comment by gyachdav** [comment]
@TheZoker  Status? can this be closed?

**Comment by TheZoker** [comment]
No not yet, because travis fails, if we undo this change. I'm in contact with the Travis support and we hopefully find the issue.

**Comment by sacdallago** [comment]
:D that's some high-end tech support and testing! Nice job

**Comment by TheZoker** [comment]
@sacdallago Seems Dominic from travis found the issue:

>  I think I know the problem. You have the following environment variable defined in your repo settings:
> $ export NODE_ENV=production
> This variable is picked up by NPM and from [the documentation](https://docs.npmjs.com/cli/install):
> With the --production flag (or when the NODE_ENV environment variable is set to production), npm will not install modules listed in devDependencies.
> Travis CI doesn't transfer environment variable in pull requests originating from forks. Hence, this explains why the build succeeded in my pull request i.e. the NODE_ENV environment variable wasn't defined in my build and the html-webpack-plugin module was properly installed from the "devDependencies" section of the package.json file.
> That being said, you might want to review if you need this environment variable or keep the html-webpack-plugin module in your "dependencies" section of your package.json file.

**Comment by TheZoker** [comment]
@mammuth @jorjo1 @mina-zaki @yashha @Andrew094 Guys I need to know, which of the [dependencies](https://github.com/Rostlab/JS16_ProjectF/blob/develop/package.json#L21) are real devDependencies, so they are not used in the main app, but for testing, linting etc.

Can you please list me these packages? I guess you know your packages best :)

**Comment by yashha** [comment]
I think only 

```
"webpack-dev-middleware": "^1.6.1",
"webpack-dev-server": "^1.14.1",
"webpack-hot-middleware": "^2.10.0"
```

They are not neccecary in 
npm run build or npm run serve

**Comment by TheZoker** [comment]
What about things like:

``````
"csslint": "^0.10.0",
"csslint-loader": "^0.2.1",
"eslint": "^2.2.0",
"eslint-loader": "^1.3.0",
"eslint-plugin-react": "^4.1.0",```
``````

**Comment by yashha** [comment]
They're used in the build process, to test while building in travis.
It would be probably better to have a separate testing env but its ok so I think.

**Comment by TheZoker** [comment]
Should be fixed with #344
